src/coffee/directives/api/models/parent/markers-parent-model.coffee

```diff
@@ -32,7 +32,7 @@ angular.module("uiGmapgoogle-maps.directives.api.models.parent")
               if (newValue != oldValue)
                 @doRebuildAll = newValue
 
-            @modelsRendered = false if not scope.models? or scope.models.length == 0
+            @modelsRendered = false if not scope.models? or @modelsLength() == 0
             @scope.$watch 'models', (newValue, oldValue) =>
```

Comment:
`@modelsLength()` checks `@scope.models?`, so checking again before calling `@modelsLength()` is redundant.

You can simplify `if not scope.models? or @modelsLength() == 0` above to `if @modelsLength() == 0`, and also `if @scope.models? and @modelsLength() > 0` below to `if @modelsLength() > 0`.